.cascade-code/Chapman.edu/test-section/tatwell/query_api_spike/query_api.vtl

```diff
@@ -0,0 +1,175 @@
+#*
+ * query_api.vtl
+ * Test format to demonstrate usage of Cascade locator tool and Query API.
+ *
+ * For more info, see spike user story:
+ * https://trello.com/c/InwKZ4w9
+ *#
+
+#import( "/_cascade/formats/helpers/debug.velocity" )
+
+##
+## Hardcoded Params
+##
+
+
+##
+## Vars
+##
+## Cascade provides a $currentPage object to every format.
+
+## This can be selected when Page Block is present:
+##set ( $pageNode = $_XPathTool.selectSingleNode($contentRoot, "//system-page[@current = 'true']") )
+
+
+##
+## Macros
+##
+#macro ( debugCurrentPage )
+  <pre>
+  # currentPage
+  currentPage name: $currentPage.name
+  currentPage path: $currentPage.path
+  currentPage assetType: $currentPage.assetType
+  currentPage Hide from navigation?: $currentPage.metadata.getDynamicField('Hide from navigation').checkbox
+
+  # _PropertyTool.outputProperties( currentPage )
+  $_PropertyTool.outputProperties( $currentPage )
+  </pre>
+#end
+
+#macro ( debugPathTools )
+  #set ( $currentPath = $currentPage.path )
+  #set ( $pathList = $currentPath.split('/') )
+  #set ( $parentIndex = $pathList.size() - 2 )
+  #set ( $parentFolderName = $pathList.get($parentIndex) )
+  #set ( $grandParentPath = $_StringTool.substringBefore($currentPath, $parentFolderName) )
+
+  <pre>
+  # debugPathTools
+  currentPage.parentFolder: $currentPage.parentFolder
+  \- isNull: $_PropertyTool.isNull($currentPage.parentFolder)
+  currentPath: $currentPath
+  pathList.size: $pathList.size()
+  parentFolderName: $parentFolderName
+  grandParentPath: $grandParentPath
+  </pre>
+#end
+
+#macro ( queryCurrentPageParentSiblings )
+  ## Note: Velocity (Java) arrays don't have a pop or remove method.
+  #set ( $parentSiblings = [] )
+
+  #set ( $currentPath = $currentPage.path )
+  #set ( $pathList = $currentPath.split('/') )
+  #set ( $parentIndex = $pathList.size() - 2 )
+  #set ( $parentFolderName = $pathList.get($parentIndex) )
```

Comment:
Will this ever throw an index-out-of-bounds type of error?

Reply:
Yes, this is good example of what the comment about needing sanity checks was referring to. I don't think it's needed for the spike but, yes, it will definitely be needed in actual implementation.